TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ultimategoal/vision/GoalFinder.java

```diff
@@ -0,0 +1,165 @@
+package org.firstinspires.ftc.teamcode.ultimategoal.vision;
+
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.opencv.core.*;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.imgproc.Moments;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opencv.imgproc.Imgproc.*;
+
+public class GoalFinder {
```

Comment:
A little misleading in that I originally thought this would detect the goal for us to aim at, but in reality it just counts how many rings for autonomous.

Reply:
it finds the angle between the robot and the goal

Reply:
please update javadoc to not say ring stacks lmao i got jebaited